### PR TITLE
Adding TotalStructuralMassProcess

### PR DESCRIPTION
**Pull request by loumalouomega**
#1422

### CODE REVIEW

**Comment by loumalouomega** [comment]
To use it just in python

~~~py
mass_process = StructuralMechanicsApplication.TotalStructuralMassProcess(model_part)
mass_process.Execute()
total_mass = model_part.ProcessInfo[NODAL_MASS]
~~~

I used NODAL_MASS to avoid add new variables, we already have to many variables

**Comment by loumalouomega** [comment]
I created the add_custom_processes_to_python, we haven't declared in the structural mechanics application

**Comment by philbucher** [comment]
small test? :)

**Comment by loumalouomega** [comment]
Fiiiiiine

**Comment by rubenzorrilla** [comment]
If there's only an Execute() method, this is in fact a utility. I would move it to the custom_utilities folder.

**Comment by loumalouomega** [comment]
I ones with only execute are precisely the processes

**Comment by rubenzorrilla** [comment]
We have precisely discussed that this morning. The Execute() method is intended to execute all the process methods. If all the implementation is in a unique method it makes no sense (have a look to your sprism_neighbours.hpp).

**Comment by loumalouomega** [comment]
I made that implementation centuries ago, I don't even know what I was doing at that time

**Comment by loumalouomega** [comment]
(This is what happen when things are not documented, the anarchy and chaos domains the PR)

**Comment by loumalouomega** [comment]
Test added

**Comment by philbucher** [comment]
why do you use `NODAL_MASS` and not just `MASS`?

**Comment by loumalouomega** [comment]
MASS is an array of three components

**Comment by loumalouomega** [comment]
PING

File: applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    
+    const std::size_t dimension = mrThisModelPart.GetProcessInfo()[DOMAIN_SIZE];
+    
+    // Now we iterate over the elements to calculate the total mass
+    ElementsArrayType& elements_array = mrThisModelPart.Elements();
+    
+    #pragma omp parallel for reduction(+:total_mass)
+    for(int i = 0; i < static_cast<int>(elements_array.size()); ++i) 
+    {
+        auto it_elem = elements_array.begin() + i;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp]: const?

File: applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        
+        // We get the condition geometry
+        GeometryType& r_this_geometry = it_elem->GetGeometry();
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp]: const?

File: applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        const std::size_t local_space_dimension = r_this_geometry.LocalSpaceDimension();
+        
+        // We get the values from the condition
+        Kratos::Properties& this_properties = it_elem->GetProperties();
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp]: const?

**Review by philbucher — COMMENTED** [review]
minor

**Comment by philbucher** [comment]
other than that you could remove some trailing whitespaces, see codacy

File: applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp (hunk 1 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp]: Huuum, not sure I will check it

File: applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp (hunk 2 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp]: Yep

File: applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp (hunk 3 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp]: Yep

**Comment by loumalouomega** [comment]
I can try, but each time after I remove the whitespaces codacy says that still there....

**Comment by loumalouomega** [comment]
http://ask.xmodulo.com/remove-trailing-whitespaces-linux.html

This solves the problem

**Review by philbucher — APPROVED** [review]
Seems fine to me

**Pull request merged**